manifests/init.pp

```diff
@@ -233,161 +202,142 @@
-  $template              = params_lookup( 'template' ),
-  $service_autorestart   = params_lookup( 'service_autorestart' , 'global' ),
-  $options               = params_lookup( 'options' ),
-  $version               = params_lookup( 'version' ),
-  $absent                = params_lookup( 'absent' ),
-  $disable               = params_lookup( 'disable' ),
-  $disableboot           = params_lookup( 'disableboot' ),
-  $monitor               = params_lookup( 'monitor' , 'global' ),
-  $monitor_tool          = params_lookup( 'monitor_tool' , 'global' ),
-  $monitor_target        = params_lookup( 'monitor_target' , 'global' ),
-  $puppi                 = params_lookup( 'puppi' , 'global' ),
-  $puppi_helper          = params_lookup( 'puppi_helper' , 'global' ),
-  $firewall              = params_lookup( 'firewall' , 'global' ),
-  $firewall_tool         = params_lookup( 'firewall_tool' , 'global' ),
-  $firewall_src          = params_lookup( 'firewall_src' , 'global' ),
-  $firewall_dst          = params_lookup( 'firewall_dst' , 'global' ),
-  $debug                 = params_lookup( 'debug' , 'global' ),
-  $audit_only            = params_lookup( 'audit_only' , 'global' ),
-  $package               = params_lookup( 'package' ),
-  $package_source        = params_lookup( 'package_source' ),
-  $package_provider      = params_lookup( 'package_provider' ),
-  $package_path          = params_lookup( 'package_path' ),
-  $service               = params_lookup( 'service' ),
-  $service_status        = params_lookup( 'service_status' ),
-  $process               = params_lookup( 'process' ),
-  $process_args          = params_lookup( 'process_args' ),
-  $process_user          = params_lookup( 'process_user' ),
-  $process_group         = params_lookup( 'process_group' ),
-  $config_dir            = params_lookup( 'config_dir' ),
-  $config_file           = params_lookup( 'config_file' ),
-  $config_file_mode      = params_lookup( 'config_file_mode' ),
-  $config_file_owner     = params_lookup( 'config_file_owner' ),
-  $config_file_group     = params_lookup( 'config_file_group' ),
-  $pid_file              = params_lookup( 'pid_file' ),
-  $data_dir              = params_lookup( 'data_dir' ),
-  $log_dir               = params_lookup( 'log_dir' ),
-  $log_file              = params_lookup( 'log_file' ),
-  $port                  = params_lookup( 'port' ),
-  $protocol              = params_lookup( 'protocol' )
-  ) inherits zabbix_agent::params {
-
-  $bool_create_user=any2bool($create_user)
-  $bool_source_dir_purge=any2bool($source_dir_purge)
-  $bool_service_autorestart=any2bool($service_autorestart)
-  $bool_absent=any2bool($absent)
-  $bool_disable=any2bool($disable)
-  $bool_disableboot=any2bool($disableboot)
-  $bool_monitor=any2bool($monitor)
-  $bool_puppi=any2bool($puppi)
-  $bool_firewall=any2bool($firewall)
-  $bool_debug=any2bool($debug)
-  $bool_audit_only=any2bool($audit_only)
-
+  $server,
```

Comment:
I prefer to set here the defaults instead of in data when they don't change according to OS.
It's more readable, also for puppet-strings